lib/srv/heartbeat.go

```diff
@@ -431,6 +431,10 @@ func (h *Heartbeat) announce() error {
 			if !ok {
 				return trace.BadParameter("expected services.Server, got %#v", h.current)
 			}
+			node, err := h.filterEC2Labels(node)
+			if err != nil {
+				return trace.Wrap(err)
+			}
```

Comment:
@EdwardDowling Why are we filtering labels on the heartbeat sending side? They should be filtered on the receiving side i.e. in the auth server's UpsertNode.